Error : mlisp89.c:508: undefined reference to `funopen'

Error : mlisp89.c:508: undefined reference to `funopen'

The funopen may be just for BSD OS.
I am on MINGW32_NT-10.0 (msys2), and run into the error case!

--- a/Makefile
+++ b/Makefile
@@ -5,11 +5,16 @@ CFLAGS += -Wall -pedantic -O2 -g
 
 all: clean mlisp89 stats test
 
+non-bsd: clean mlisp89-non-bsd stats test
+
 micro-lisp: micro-lisp.c
 	$(CC) $(CFLAGS) -o $@ $^
 
 mlisp89: mlisp89.c
 	$(CC) $(CFLAGS) -ansi -std=c89 -D_DEFAULT_SOURCE -o $@ $^
+
+mlisp89-non-bsd: mlisp89.c
+	$(CC) $(CFLAGS) -ansi -lbsd -std=c89 -D_DEFAULT_SOURCE -o mlisp89 $^
 
 stats: mlisp89.c
 	wc $^

--- a/mlisp89.c
+++ b/mlisp89.c
@@ -1,6 +1,12 @@
 /* Micro Scheme (C) A. Carl Douglas */
 #include <stdint.h>
+
+#if defined(BSD)
 #include <stdio.h>
+#else
+#include <bsd/stdio.h>
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 #define debug(m,e) fprintf(default_output_port->value.stream, "%s:%d: %s:",__FILE__,__LINE__,m); print_obj(default_output_port, e); fprintf(default_output_port->value.stream, "\n");

